scripts/update-test-tsconfig.js

```diff
@@ -12,18 +15,17 @@ const CONDITIONAL_FILES = {
  * @param {string} tsVersion
  */
 const applyRules = (json, tsVersion) => {
-  const ALL_CONDITIONAL_FILES = Object.values(CONDITIONAL_FILES);
+  const ALL_CONDITIONAL_FILES = Object.keys(CONDITIONAL_FILES);
 
   // reset `exclude` to the original state, without any conditional files
 
   let exclude = (json.exclude || []).filter((v) => !ALL_CONDITIONAL_FILES.includes(v));
 
-  if (["4.5", "4.6", "4.7", "4.8"].includes(tsVersion)) {
-    // XOR utility type never inferred the right types in functions parameters
-    // for TypeScript versions between 4.5.x and 4.8.x
-
-    exclude.push(CONDITIONAL_FILES.XOR_FUNCTION_PARAMETERS);
-  }
+  Object.entries(CONDITIONAL_FILES).forEach(([file, version]) => {
+    if (semver.satisfies(tsVersion, version)) {
```

Comment:
I suggest renaming a `version` to `tsVersionConstraint`, because it may contain multiple versions, a range, etc

```suggestion
  Object.entries(CONDITIONAL_FILES).forEach(([file, tsVersionConstraint]) => {
    if (semver.satisfies(tsVersion, tsVersionConstraint)) {
```

WDYT?

Reply:
Yeah, makes sense!